tests/main/interfaces-alsa/task.yaml

```diff
@@ -0,0 +1,97 @@
+summary: Ensure that the alsa interface works.
+
+details: |
+    The alsa interface allows connected plugs to access raw ALSA devices.
+
+    A snap which defines a alsa plug must be shown in the interfaces list.
+    The plug must not be autoconnected on install and, as usual, must be able to
+    be reconnected.
+
+    A snap declaring a plug on this interface must be able to access to raw ALSA
+    devices, for this test we check the low level security rules by creating
+    specific devices under /dev/snd, a cgroup if needed and the ALSA state dir,
+    exercising in each caase the read or read-write permissions that must be in
+    place.
+
+prepare: |
+    echo "Given a snap declaring a plug on the alsa interface is installed"
+    . $TESTSLIB/snaps.sh
+    install_generic_consumer alsa
+
+restore: |
+    rm -f *.error /dev/snd/mysnd-dev
+
+execute: |
+    CONNECTED_PATTERN=":alsa +generic-consumer"
+    DISCONNECTED_PATTERN="\- +generic-consumer:alsa"
+
+    echo "The plug is not connected by default"
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "==================================="
+
+    echo "When the plug is connected"
+    snap connect generic-consumer:alsa
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    echo "Then the snap is able to access snd devices"
+    generic-consumer.cmd touch /dev/snd/mysnd-dev
+    echo "mysnd-dev-content" | tee /dev/snd/mysnd-dev
+    generic-consumer.cmd cat /dev/snd/mysnd-dev | MATCH mysnd-dev-content
+    generic-consumer.cmd rm /dev/snd/mysnd-dev
+
+    echo "And the snap is able to access the related udev data"
+    if [ ! -f /run/udev/data/c116:0 ]; then
+        touch /run/udev/data/c116:0
+        echo "myudevdata-content" | tee /run/udev/data/c116:0
+        generic-consumer.cmd cat /run/udev/data/c116:0 | MATCH myudevdata-content
+        rm /run/udev/data/c116:0
+    fi
+
+    # TODO: extend test to check /var/lib/alsa with plug connected when
+    # https://bugs.launchpad.net/snapd/+bug/1694281 is fixed
+
+    if [ "$(snap debug confinement)" = none ] ; then
+        echo "Do not execute checks with disconnected plug on systems where confinement doesn't work"
+        exit 0
+    fi
+
+    echo "==================================="
+
+    echo "When the plug is disconnected"
+    snap disconnect generic-consumer:alsa
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    echo "The snap is not able to access snd devices"
+    if generic-consumer.cmd touch /dev/snd/mysnd-dev 2>snd-create.error; then
+        echo "Create snd device with disconnected plug should fail"
+        exit 1
+    fi
+    MATCH "Permission denied" ./snd-create.error
```

Comment:
That won't work at the moment. If the matching fails, it won't print the file content.

That said, let's leave it as is and fix MATCH in spread instead.